File: data/src/main/kotlin/dev/yashgarg/kimai/models/Customer.kt
/*
 * Copyright © 2023 Yash Garg.
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://opensource.org/licenses/MIT.
 */
package dev.yashgarg.kimai.models

import androidx.annotation.Keep
import kotlinx.serialization.Serializable

@Keep
@Serializable
data class Customer(
  val id: Int,
  val name: String,
  val number: String,
  val comment: String? = null,
  val visible: Boolean,
  val billable: Boolean,
  val metaFields: List<MetaField>,
  val teams: List<Team>,
  val color: String? = null
)

File: lib/data/models/customer.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:kimai/data/models/models.dart';

part '../../generated/data/models/customer.freezed.dart';
part '../../generated/data/models/customer.g.dart';

@freezed
class Customer with _$Customer {
  const factory Customer({
    required int id,
    required String name,
    required String number,
    String? comment,
    required bool visible,
    required bool billable,
    required List<MetaField> metaFields,
    required List<Team> teams,
    String? color,
  }) = _Customer;

  factory Customer.fromJson(Map<String, dynamic> json) =>
      _$CustomerFromJson(json);
}
